control/main.py

```diff
@@ -0,0 +1,232 @@
+    data[u'x'] = x
+    data[u'y'] = y
+    data[u'z'] = location.alt
+    i2c_data = data_client.read()
+    data[u'temp'] = float(i2c_data['temperature'])
+    data[u'lat'] = location.lat
+    data[u'lon'] = location.lon
+    data[u'time'] = flight_time
+    return data
+
+
+def main():
+    """Takes the drone up and then lands."""
+    # Setup logging
+    logger = logging.getLogger('control')
+    logger.setLevel(logging.DEBUG)
+    filehandler = logging.FileHandler('main.log')
+    filehandler.setLevel(logging.DEBUG)
+    console = logging.StreamHandler()
+    console.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    filehandler.setFormatter(formatter)
+    console.setFormatter(formatter)
+    logger.addHandler(filehandler)
+    logger.addHandler(console)
+
+    # Connect to xBee
+    com = Communication(COM_CONNECTION_STRING, 0.1)
+    logger.debug("Connected to wireless communication receiver")
+    com.send(u"Connected to wireless communication receiver")
+
+    # Connect to i2c data server
+    data_client = I2cDataClient("tcp://localhost:5555")
+    if not data_client.read():
+        logger.critical("Can't connect to zmq data server")
+        com.send(u"Can't connect to zmq data server")
+        sys.exit(1)
+
+    # Connect to the drone
+    logger.debug("Starting program, attempting connection to flight controller.")
+    com.send(u"Starting program, attempting connection to flight controller")
+    vehicle_control = None
+    try:
+        vehicle_control = Controller(PIXHAWK_CONNECTION_STRING, baud=57600, com=com)
+        logger.debug("Connected to flight controller")
+        com.send(u"Connected to flight controller")
+    except dronekit.APIException as err:
+        logger.critical("dronekit.APIException: {}".format(err))
+        logger.critical("Could not connect to flight controller.")
+        com.send(u"Could not connect to flight controller.")
+        sys.exit(1)
+
+    # Wait until the waypoints flight path is received from GCS
+    logger.debug("Waiting to receive flight path from GCS")
+    com.send(u"Waiting to receive flight path from GCS")
+    waypoints = com.receive()
+    while not waypoints:
+        waypoints = com.receive()
+        time.sleep(1)
+
+    # Create points
+    start_location = vehicle_control.vehicle.location.global_relative_frame
+    points = create_waypoints(logger, com, start_location, waypoints)
+
+    if not points:
+        logger.critical("Invalid points received from GCS")
+        com.send(u"Invalid points received from GCS")
+        sys.exit(1)
+
+    # Arm and takeoff
+    vehicle_control.arm()
+    vehicle_control.takeoff(10)
+    points.append(vehicle_control.home)
+
+    # Log points
+    for index, point in enumerate(points):
+        logger.debug("Destination {}: {}".format(index, point))
+
+    # Go to the points
+    flight_start_time = time.time()
```

Comment:
Assuming you know this cuz it makes sense to do it this way, but just fyi this makes start time equal to when the drone reaches its target altitude. Since this is when we first log points it makes sense, but wasn't aware if you realized that or not.